plugins/logitech-hidpp/fu-logitech-hidpp-bootloader.c

```diff
@@ -232,15 +234,20 @@ fu_logitech_hidpp_bootloader_set_bl_version (FuLogitechHidPpBootloader *self, GE
 	 * 012345678901234 */
 	build = (guint16) fu_logitech_hidpp_buffer_read_uint8 ((const gchar *) req->data + 10) << 8;
 	build += fu_logitech_hidpp_buffer_read_uint8 ((const gchar *) req->data + 12);
-	version = fu_logitech_hidpp_format_version ("BOT",
-			fu_logitech_hidpp_buffer_read_uint8 ((const gchar *) req->data + 3),
-			fu_logitech_hidpp_buffer_read_uint8 ((const gchar *) req->data + 6),
-			build);
+	major = fu_logitech_hidpp_buffer_read_uint8 ((const gchar *) req->data + 3);
+	minor = fu_logitech_hidpp_buffer_read_uint8 ((const gchar *) req->data + 6);
+	version = fu_logitech_hidpp_format_version ("BOT", major, minor, build);
 	if (version == NULL) {
 		g_prefix_error (error, "failed to format firmware version: ");
 		return FALSE;
 	}
 	fu_device_set_version_bootloader (FU_DEVICE (self), version);
+
+	if ((major == 0x01 && minor >= 0x04) ||
+	    (major == 0x03 && minor >= 0x02))
+		fu_device_set_protocol (FU_DEVICE (self), "com.logitech.unifyingsigned");
```

Comment:
Wouldn't it be better to change the protocol to be defined as `com.logitech.{nordic,texas}.{un,}signed`? We have all the required information right here. I don't think there's any need to have the split this way.

Reply:
Well right now there isn't actually any protocol prefixes defined as nordic or texas.  We also don't have fallback code to allow devices to have multiple protocols.  For now this matches the protocols in use by the stack and LVFS.

I think we should work out the details of introducing more protocols (and potentially fallback protocols) separately from this bug fix (as discussing in https://github.com/fwupd/fwupd/issues/1876)